op-batcher/batcher/channel_manager.go

```diff
@@ -139,9 +142,13 @@ func (s *channelManager) removePendingChannel(channel *channel) {
 // nextTxData dequeues frames from the channel and returns them encoded in a transaction.
 // It also updates the internal tx -> channels mapping
 func (s *channelManager) nextTxData(channel *channel) (txData, error) {
-	if channel == nil || !channel.HasTxData() {
+	if channel == nil {
+		s.log.Trace("no channel")
+		return txData{}, io.EOF
```

Comment:
note that you still return the `io.EOF` error here, not the new error type.

Reply:
Reverted to the old behaviour [510464d](https://github.com/ethereum-optimism/optimism/pull/12002/commits/510464d28cda1e5f0c2690be810ddda5c24a68c0)